### PR TITLE
Improve error handling of ajax errors when providing form schema

### PR DESCRIPTION
**Pull request by tractorcow**
Parent issue: https://github.com/silverstripe/silverstripe-asset-admin/issues/528

Requires https://github.com/silverstripe/silverstripe-admin/pull/198

### CODE REVIEW

File: client/src/containers/Editor/Editor.js [hunk 1]
```diff
@@ -152,14 +152,12 @@ class Editor extends Component {
 
     let error = null;
     if (this.state.loadingError) {
-      const message = this.state.loadingError.value;
+      let message = this.state.loadingError.value;
+      if (this.state.loadingError.code === 404) {
+        message = i18n._t('AssetAdmin.FILE_MISSING', 'File cannot be found');
+      }
```
Comment (flamerohr) [inline client/src/containers/Editor/Editor.js]: how about one step further and adding an `Unknown error occurred` message for when `message` is empty

Comment (flamerohr) [inline client/src/containers/Editor/Editor.js]: Oh, I see it on the FormBuilderLoader end... Might pay to have something on this end as well, as loadingError may be handled by something else in future

File: client/src/containers/Editor/Editor.js [hunk 2]
```diff
@@ -156,6 +156,9 @@ class Editor extends Component {
       if (this.state.loadingError.code === 404) {
         message = i18n._t('AssetAdmin.FILE_MISSING', 'File cannot be found');
       }
+      if (!message) {
+        message = i18n._t('AssetAdmin.UNKNOWN_ERROR', 'An unknown error has occurred');
```
Comment (tractorcow) [inline client/src/containers/Editor/Editor.js]: Shall we use the existing Admin.UNKNOWN_ERROR key?

Comment (flamerohr) [inline client/src/containers/Editor/Editor.js]: yeah, could do with the explicit dependency on Admin

File: code/GraphQL/FileFilterInputTypeCreator.php [hunk 3]
```diff
@@ -88,6 +89,13 @@ public function filterList(Filterable $list, $filter)
         // ID filtering
         if (isset($filter['id']) && (int)$filter['id'] > 0) {
             $list = $list->filter('ID', $filter['id']);
+            
+            if ($list->count() === 0) {
+                throw new HTTPResponse_Exception(_t(
+                    __CLASS__ . 'FileNotFound',
```
Comment (tractorcow) [inline code/GraphQL/FileFilterInputTypeCreator.php]: String being concatenated needs to have a leading .

```diff
- __CLASS__ . 'FileNotFound',
+ __CLASS__ . '.FileNotFound',
`

File: client/src/state/files/readFilesQuery.js [hunk 4]
```diff
@@ -113,11 +114,16 @@ const config = {
 
     const filesLoading = (folder && !folder.children);
 
+    const errors = error && error.graphQLErrors &&
+      error.graphQLErrors.map((graphQLError) => (
+        `${graphQLError.message} ${JSON.stringify(graphQLError.locations)}`
```
Comment (tractorcow) [inline client/src/state/files/readFilesQuery.js]: The `.locations` component doesn't seem to be useful to the user. Shall we omit it?

Comment (flamerohr) [inline client/src/state/files/readFilesQuery.js]: hmm yeah, that locations data seems to be quite useless for devs as well (at least until it stops masking everything)

**Pull request merged**